app/models/runtime/role.rb

```diff
@@ -5,99 +5,199 @@ module VCAP::CloudController
 
   # Sequel allows to create models based on datasets. The following is a dataset that unions all the individual roles
   # tables and labels each row with a `type` column based on which table it came from
-  class Role < Sequel::Model(
-    OrganizationUser.select(
-      Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_USER, :type),
-      Sequel.as(:role_guid, :guid),
-      :user_id,
-      :organization_id,
-      Sequel.as(SPACE_OR_ORGANIZATION_NOT_SPECIFIED, :space_id),
-      :created_at,
-      :updated_at
-    ).union(
-      OrganizationManager.select(
-        Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_MANAGER, :type),
-        Sequel.as(:role_guid, :guid),
-        :user_id,
-        :organization_id,
-        Sequel.as(SPACE_OR_ORGANIZATION_NOT_SPECIFIED, :space_id),
-        :created_at,
-        :updated_at),
-      all: true,
-      from_self: false
-    ).union(
-      OrganizationBillingManager.select(
-        Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_BILLING_MANAGER, :type),
-        Sequel.as(:role_guid, :guid),
-        :user_id,
-        :organization_id,
-        Sequel.as(SPACE_OR_ORGANIZATION_NOT_SPECIFIED, :space_id),
-        :created_at,
-        :updated_at),
-      all: true,
-      from_self: false
-    ).union(
-      OrganizationAuditor.select(
-        Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_AUDITOR, :type),
+
+  def self.role_dataset(organization_id=false)
+    if organization_id
+      spaces_in_org = Space.where(organization_id: organization_id).map(&:id)
```

Comment:
is it necessary to pull the `id`s into memory?  Usually pulling IDs into memory is slower than constructing a dataset (see https://github.com/cloudfoundry/cloud_controller_ng/pull/3070 for an example).

Reply:
I wonder if it is better in this case because it guarantees we execute the query once, instead of this being evaluated as a sub-query each time.  Not sure if the DB will cache the results for multiple identical subqueries or something.  Might be worth testing.